src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -333,23 +333,44 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 
 	private showTaskNotification(): void {
 		const storageKey = 'workbench.tasks.ranTaskBefore';
-		if (!this.storageService.get(storageKey)) {
+		if (this.storageService.get(storageKey)) {
 			const fileName = path.relative(this.contextService.getWorkspace().resource.toString(), this.resource.toString());
 			if (fileName.match(/^gruntfile\.js$/i) || fileName.match(/^gulpfile\.js$/i) || fileName.match(/^tsconfig\.json$/i)) {
-				const message = localize('taskFileOpened', "Visual Studio Code has extra functionality for this type of file");
-				const action = this.instantiationService.createInstance(ShowTasksAction, ShowTasksAction.ID, localize('showTasks', "Show Tasks"));
-
+				const message = localize('taskFileOpened', `Run your ${fileName.split('.')[0]} in vscode. Get started here.`);
+				let action: any;
+				let messageTest: string;
+				const showDocumentation = isShowTaskDocumentation(this.storageService);
+				if (showDocumentation) {
+					action = this.instantiationService.createInstance(ShowTasksDocumentationAction, ShowTasksDocumentationAction.ID, localize('showTaskDocumentation', "Show task Documentation"));
```

Comment:
Code style wise, this is totally fine, even though it feels weird to declare action and messageTest above then duplicate the assignments like this. But JS doesn't really give you better tools and this is one way I would write it.